tests/ZfcRbacTest/Factory/ControllerPermissionsGuardFactoryTest.php

```diff
@@ -43,20 +49,51 @@ public function testFactory()
             'protection_policy' => GuardInterface::POLICY_ALLOW,
         ]);
 
-        $serviceManager = new ServiceManager();
         $serviceManager->setService('ZfcRbac\Options\ModuleOptions', $options);
         $serviceManager->setService(
             'ZfcRbac\Service\AuthorizationService',
             $this->getMock('ZfcRbac\Service\AuthorizationService', [], [], '', false)
         );
 
-        $pluginManager = new GuardPluginManager();
-        $pluginManager->setServiceLocator($serviceManager);
+        $pluginManager = new GuardPluginManager($serviceManager);
```

Comment:
yes it would, but as i said, the factories have BC break, if you extended or aggregated them, you might need to update your code anyway. this way we can ensure the "new way" of injecting via constructor is used everywhere.